### PR TITLE
[TAS-880] ✨ Support storing NFT Book in API wallet

### PR DESCRIPTION
**Pull request by WeiJun0827**
- Add isAutoDeliver boolean in each price(edition)
- User can set isAutoDeliver when creating an NFT Book or adding a new price
- isAutoDeliver is read-only after created
- Stock of auto delivering price can not be modified

Depends on https://github.com/likecoin/likecoin-api-public/pull/589

### CODE REVIEW

File: pages/nft-book-store/new.vue [hunk 1]
```diff
@@ -655,6 +665,7 @@ function mapPrices (prices:any) {
       priceInDecimal: Math.round(Number(p.price) * 100),
       price: Number(p.price),
       stock: Number(p.stock),
+      isAutoDeliver: p.deliverMethod === 'auto',
```
Comment (williamchong) [inline pages/nft-book-store/new.vue:668]: Please allow setting a default memo for `isAutoDeliver`

File: pages/nft-book-store/new.vue [hunk 2]
```diff
@@ -712,6 +723,37 @@ async function submitNewClass () {
...
+      })
+      const nftIds = nfts.map(nft => nft.id).slice(0, autoDeliverCount)
+
+      const res = await signSendNFTs(
+        LIKER_NFT_TARGET_ADDRESS,
+        classIdInput.value as string,
+        nftIds,
+        signer.value,
+        wallet.value,
+        'Send auto delivered NFT to API wallet'
```
Comment (williamchong) [inline pages/nft-book-store/new.vue:749]:
```suggestion
        'Send auto delivered NFT Book to API wallet'
```

File: pages/nft-book-store/status/[classId]/edit/[editionIndex].vue [hunk 3]
```diff
@@ -350,6 +359,7 @@ async function handleSubmit () {
       priceInDecimal: Math.round(Number(price.value) * 100),
       price: Number(price.value),
       stock: Number(stock.value),
+      isAutoDeliver: deliverMethod.value === 'auto',
```
Comment (williamchong) [inline pages/nft-book-store/status/[classId]/edit/[editionIndex].vue:362]: Please allow setting a default memo for `isAutoDeliver`

File: pages/nft-book-store/status/[classId]/edit/new.vue [hunk 4]
```diff
@@ -338,6 +346,7 @@ async function handleSubmit () {
       priceInDecimal: Math.round(Number(price.value) * 100),
       price: Number(price.value),
       stock: Number(stock.value),
+      isAutoDeliver: deliverMethod.value === 'auto',
```
Comment (williamchong) [inline pages/nft-book-store/status/[classId]/edit/new.vue:349]: Please allow setting a default memo for `isAutoDeliver`

Comment (williamchong) [inline pages/nft-book-store/status/[classId]/edit/new.vue:349]: How does one send NFTs for this new price?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: utils/cosmos.ts [hunk 5]
```diff
@@ -246,7 +246,7 @@ export async function signSendNFT (
     address,
     targetAddress,
     classId,
-    [nftId],
+    nftIds,
     { memo }
   ) as DeliverTxResponse
   return res
```
Comment (WeiJun0827) [inline utils/cosmos.ts:252]: @williamchong seems we are going to have two `signSendNFTs`, where one sends multiple classId-nftId pairs with a one-to-one relationship for collection feature, and another for sending multiple nftIds with the same classId.
Do you have any suggestions on naming or reusing the function?

Comment (williamchong) [reply utils/cosmos.ts:252]: How about keep repeating the same classId in the 1-to-1 version?

**Review by williamchong — APPROVED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**